subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc

```diff
@@ -0,0 +1,849 @@
+
+[source,xml]
+----
+<component group="org.apache.pdfbox" name="pdfbox" version="2.0.17">
+   <artifact name="pdfbox-2.0.17.jar">
+      <md5 value="c713a8e252d0add65e9282b151adf6b4" origin="official site"/>
+      <sha1 value="b5c8dff799bd967c70ccae75e6972327ae640d35" origin="official site"/>
+   </artifact>
+</component>
+----
+
+There are multiple reasons why you'd like to do so:
+
+1. an official site doesn't publish _secure_ checksums (SHA-256, SHA-512) but publishes multiple insecure ones (MD5, SHA1). While it's easy to fake a MD5 checksum and hard but possible to fake a SHA1 checksum, it's harder to fake both of them for the same artifact.
+2. you might want to add generated checksums to the list above
+3. when _updating_ dependency verification file with more secure checksums, you don't want to accidentally erase checksums
+
+[[sec:signature-verification]]
+== Verifying dependency signatures
+
+In addition to <<sec:checksum-verification,checksums>>, Gradle supports verification of signatures.
+Signatures are used to assess the _provenance_ of a dependency (it tells who signed the artifacts, which usually corresponds to who produced it).
+They are also used to assess the integrity of a dependency just like checksums.
+While enabling signature verification usually means a higher level of security, you might want to replace checksum verification with signature verification.
+
+However:
+
+- Gradle only supports verification of signatures published on remote repositories as ASCII-armored PGP files
+- Not all artifacts are published with signatures
+
+As a consequence, signature verification will often be used alongside checksum verification.
+
+=== Enabling signature verification
+
+Because verifying signatures is more expensive (both I/O and CPU wise) and harder to check manually, it's not enabled by default.
+
+Enabling it requires you to change the configuration option in the `verification-metadata.xml` file:
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<verification-metadata>
+   <configuration>
+      <verify-signatures>true</verify-signatures>
+   </configuration>
+</verification-metadata>
+----
+
+[[sec:understanding-signature-verification]]
+== Understanding signature verification
+
+Once signature verification is enabled, for each artifact, Gradle will:
+
+* try to download the corresponding `.asc` file
+* if it's present
+** automatically download the keys required to perform verification of the signature
+** verify the artifact using the downloaded public keys
+* if it's absent, fallback to checksum verification
+
+That is to say that Gradle verification mechanism is much stronger if signature verification is enabled than just with checksum verification.
+In particular:
+
+- if an artifact is signed with multiple keys, all of them must pass validation or the build will fail
+- if an artifact passes verification, checksums are _ignored_
```

Comment:
Oh. That is unexpected.

If users want to ignore checksums, they can just remove (or comment) the relevant bits of the configuration file. Otherwise, it would look like the checksums are verified while in practice they are ignored :(

Reply:
I think it makes sense, changing this to _also_ verify checksums is easy.

Reply:
Done in #11756